content/blog/2021/07/2021-07-27-git-credentials-binding-phase-1.adoc

```diff
@@ -0,0 +1,89 @@
+---
+:layout: post
+:title: "Git username / password credentials binding"
+:tags:
+- git
+authors:
+- markewaite
+- rishabhbudhouliya
+links:
+  discourse: https://community.jenkins.io/t/git-username-password-credentials-binding-has-released/263
+description: >
+  In this article we announce the release of Git username / password credentials binding as part of git plugin 4.8.0.
+  Thanks to Google Summer of Code for sponsoring the project and special thanks to Harshit Chopra for the implementation.
+opengraph:
+  image: /images/post-images/2021-07-git-credentials-binding-phase-1/git-credentials-binding-phase-1-opengraph.png
+---
+
+image:/images/post-images/2021-07-git-credentials-binding-phase-1/git-credentials-binding-phase-1-opengraph.png[Git username/password credentials binding, role=center, link="https://plugins.jenkins.io/git/"]
+
+Google Summer of Code 2021 is implementing link:/projects/gsoc/2021/projects/git-credentials-binding/[git credentials binding for `sh`, `bat`, and `powershell`].
+Git credentials binding is one of the most requested features for Jenkins Pipeline (see jira:JENKINS-28335[]).
+
+The project involves extending the Credentials Binding Plugin to create custom bindings for two types of credentials essential to establish a remote connection with a git repository
+
+* Username/Password
+* SSH Private Key
+
+== Why use git credentials binding?
+
+Many operations in a Jenkins Pipeline or Freestyle job can benefit from authenticated access to git repositories.
+Authenticated access to a git repository allows a Jenkins job to
+
+* apply a tag and push the tag
+* merge a commit and push the merge
+* update submodules from private repositories
+* retrieve large files with git LFS
+
+The git credentials username / password binding included in git plugin 4.8.0 allows Pipeline and Freestyle jobs to use command line git from `sh`, `bat`, and `powershell` for authenticated access to git repositories.
+
+== How to use git credentials binding?
+
+The binding is accessible using the `withCredentials` Pipeline step.
+It requires two parameters:
+
+* `credentialsId`: Reference id provided by creating a Username/Password type credential in the Jenkins configuration. To understand how to configure credentials in a Jenkins environment: link:/doc/book/using/using-credentials.adoc[Using Credentials]
```

Comment:
Need to not include the adoc suffix, otherwise the link is broken
```suggestion
* `credentialsId`: Reference id provided by creating a Username/Password type credential in the Jenkins configuration. To understand how to configure credentials in a Jenkins environment: link:/doc/book/using/using-credentials[Using Credentials]
```